tensorboard/components_polymer3/vz_line_chart2/vz-line-chart2.ts

```diff
@@ -139,58 +144,157 @@ class VzLineChart2 extends LegacyElementMixin(PolymerElement) {
     </style>
   `;
 
+  /**
+   * Scale that maps series names to colors. The default colors are from
+   * d3.schemeCategory10. Use this property to replace the default line
+   * colors with colors of your own choice.
+   */
   @property({type: Object})
-  colorScale: object = new Plottable.Scales.Color().range(
+  colorScale: Plottable.Scales.Color = new Plottable.Scales.Color().range(
     d3.schemeCategory10.slice(0)
   );
 
+  /**
+   * A function that takes a data series string and returns a
+   * Plottable.SymbolFactory to use for rendering that series. This property
+   * implements the vz_chart_helpers.SymbolFn interface.
+   */
   @property({type: Object})
-  symbolFunction: object;
+  symbolFunction: SymbolFn;
 
+  /**
+   * Whether smoothing is enabled or not. If true, smoothed lines will be
+   * plotted in the chart while the unsmoothed lines will be ghosted in
+   * the background.
+   */
   @property({
     type: Boolean,
     notify: true,
   })
   smoothingEnabled: boolean = false;
 
+  /**
+   * Weight (between 0.0 and 1.0) of the smoothing. A value of 0.0
+   * means very little smoothing, possibly no smoothing at all. A
+   * value of 1.0 means a whole lot of smoothing, possibly so much as
+   * to make the whole plot appear as a constant function.
+   *
+   * Has no effect when `smoothingEnabled` is `false`.
+   */
   @property({type: Number})
   smoothingWeight: number = 0.6;
 
+  /**
+   * This is a helper field for automatically generating commonly used
+   * functions for xComponentsCreationMethod. Valid values are what can
+   * be processed by vz_chart_helpers.getXComponents() and include
+   * "step", "wall_time", and "relative".
+   */
   @property({type: String})
-  xType: string = '';
+  xType: XType = XType.STEP;
```

Comment:
Before, a falsy default tells `_makeChart` to use `xComponentsCreationMethod` passed in.
By having a truthy default, will this change the behavior and always override the `xComponentsCreationMethod`?

Reply:
Changed it back to `null` for falsey.